pkg/sql/schemachanger/scplan/internal/rules/dep_index_and_column.go

```diff
@@ -358,37 +383,45 @@ func init() {
 // TODO(postamar): express this rule in a saner way
 func init() {
 
-	depRule(
+	registerDepRule(
 		"column type removed right before column when not dropping relation",
 		scgraph.SameStagePrecedence,
-		scpb.ToAbsent,
-		element(scpb.Status_ABSENT,
-			(*scpb.ColumnType)(nil),
-		),
-		element(scpb.Status_ABSENT,
-			(*scpb.Column)(nil),
-		),
-		screl.DescID,
-		screl.ColumnID,
-	).withFilter("parent-relation-is-not-dropped", func(ct *scpb.ColumnType, _ *scpb.Column) bool {
-		return !ct.IsRelationBeingDropped
-	}).register()
+		"column-type", "column",
+		func(from, fromTarget, fromNode, to, toTarget, toNode rel.Var) rel.Clauses {
+			return rel.Clauses{
+				from.Type((*scpb.ColumnType)(nil)),
+				to.Type((*scpb.Column)(nil)),
+				joinOnColumnID(from, to),
+				targetStatusEq(fromTarget, toTarget, scpb.ToAbsent),
+				currentStatusEq(fromNode, toNode, scpb.Status_ABSENT),
+				rel.Filter("columnTypeIsNotBeingDropped", from)(func(
+					ct *scpb.ColumnType,
+				) bool {
+					return !ct.IsRelationBeingDropped
+				}),
+			}
+		},
+	)
 
-	depRule(
+	registerDepRule(
 		"partial predicate removed right before secondary index when not dropping relation",
 		scgraph.SameStagePrecedence,
-		scpb.ToAbsent,
-		element(scpb.Status_ABSENT,
-			(*scpb.SecondaryIndexPartial)(nil),
-		),
-		element(scpb.Status_ABSENT,
-			(*scpb.SecondaryIndex)(nil),
-		),
-		screl.DescID,
-		screl.IndexID,
-	).withFilter("parent-relation-is-not-dropped", func(ip *scpb.SecondaryIndexPartial, _ *scpb.SecondaryIndex) bool {
-		return !ip.IsRelationBeingDropped
-	}).register()
+		"partial-predicate", "index",
+		func(from, fromTarget, fromNode, to, toTarget, toNode rel.Var) rel.Clauses {
+			return rel.Clauses{
+				from.Type((*scpb.SecondaryIndexPartial)(nil)),
+				to.Type((*scpb.SecondaryIndex)(nil)),
+				joinOnIndexID(from, to),
+				targetStatusEq(fromTarget, toTarget, scpb.ToAbsent),
+				currentStatusEq(fromNode, toNode, scpb.Status_ABSENT),
+				rel.Filter("secondaryIndexPartialIsNotBeingDropped", from)(func(
```

Comment:
Kebab-case all the rel things?

Reply:
I made the filter functions camelCase. I'm starting to dislike the kebabs. If I were typing this language I would not be pleased with kabobs I don't think. Sometimes I think about what an interpreter would look like. I'm going to keep iterating around these parts, happy to make further changes later.